Sage/carbon
# fix(pager): change styles so all items are exposed to accessibility tree

When using Chrome, when the Previous and Next buttons are missing due to only 1 page being present,
screenreaders such as NVDA and VoiceOver were treating "items" and "page", and the total page number
and "X items" messages, as the same element when moving through one item at a time. This appears to
be due to a bug or feature in Chrome whereby generic elements (div or span) with display property
inline or flex were not being exposed to the accessibility tree at all. So "X items" has been
changed from flex to block, and "page" from inline to block (div in place of span), to fix this.

fix #4637

### Proposed behaviour

When using the NVDA screenreader on Windows, and navigating one page at a time, when a Pager is displayed but missing the "previous" and "next" buttons, the word "items" (next to the select) should be read separately from "page", and the numbers in "page 1 and 1" and "2 items" should be read separately from each other.



### Current behaviour

When using Chrome, both NVDA on Windows and Voiceover on Chrome read the above two items as one when stepping through one element at a time - so you hear "items-page" and "1-2 items" at the same time, which is illogical and confusing. It's particularly bad on NVDA as there the "1-2items" is actually read as "12 items".



### Checklist




#### QA


### Additional context



### Testing instructions

Can be tested on this Storybook story: https://carbon.sage.com/?path=/story/flat-table--with-sticky-footer-inside-of-larger-div

Test in all supported browser/screenreader combinations on both Windows and Mac, stepping through one item at a time (shortcut-keys: Control+Option+RightArrow on VO, just right arrow on NVDA), to ensure the behaviour is as expected.



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
